### PR TITLE
[REF] Angular - cleanup core angular module declarations

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This fixes a long-standing FIXME and simplifies the loading of core Angular modules.

Before
----------------------------------------
1. Core angular-modules loaded from a hand-curated list.
2. CiviMail and CiviCase angular modules loaded from a component callback marked "experimental"

After
----------------------------------------
1. Core angular-modules auto-loaded.
2. Component callback removed. I guess the experiment is over.

Technical Details
----------------------------------------
Moved the CiviCase and CiviMail modules into their respective core-extension `ang/` directories so they are now autoloaded in the normal way.

### CODE REVIEW

File: CRM/Core/Component/Info.php [hunk 1]
```diff
@@ -110,17 +110,6 @@ public function __construct($name, $namespace, $componentID) {
     $this->info['url'] = $this->getKeyword();
   }
 
-  /**
-   * EXPERIMENTAL: Get a list of AngularJS modules
```
Comment (colemanw) [inline CRM/Core/Component/Info.php:114]: @totten I guess the experiment is over

**Comment by seamuslee001** [comment]
@colemanw Karma test fail relates here I am guessing that before running Karma we aren't enabling CiviCase Component/Extension right?

```
Error: [$injector:modulerr] Failed to instantiate module crmCaseType due to:
Error: [$injector:nomod] Module 'crmCaseType' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
```

File: CRM/Core/Component/Info.php (hunk 1 above)
Comment (totten) [reply CRM/Core/Component/Info.php:114]: lol

**Comment by totten** [comment]
Well, that definitely looks prettier.

PrettyPicture


To kick the tires, I tried a few things:

* On a demo site, enable/disable some components (CiviCase/CiviMail). Visit their respective paths under `civicrm/a`. Use "View Source" to see what modules were loaded. This all behaved as expected.
* On CLI, make a file-dump with metadata about Angular modules. Compare the metadata before+after the patch.
    ```
    git checkout master
    cv flush && cv ang:html:list > /tmp/before-html.txt && cv ang:module:list > /tmp/before-mods.txt

    git checkout angularGlob
    cv flush && cv ang:html:list > /tmp/after-html.txt && cv ang:module:list > /tmp/after-mods.txt

    colordiff -u /tmp/{before,after}-html.txt
    colordiff -u /tmp/{before,after}-mods.txt
    ```

    It was important to flush caches in between the changes, which I think is fine. The results matched up.
* I was curious about the fact that the declared `ext` would become different (`ext=>civicrm` becomes `ext=>civi_case`). But after skimming references to `ext`, I can't see any problems with it.
    * The `ext` is used internally to figure out the full file-path/file-URL. It's good+proper for this to change (since the file-path has changed).
    * The `ext` is not used as selector in hook contexts. (Well, at least `hook_alterAngular` targets partials by their logical path -- which is still the same -- as demonstrated by `cv ang:html:list`)    * The `ext` is used internally for translations. But the effective here would be additive. (When it was `ext=>civicrm`, it would only check `civicrm.mo`. Now that it moved to `ext=>civi_case`, it could theoretically consult multiple MOs `[civi_case.mo,civicrm.mo]`. Either way it reads `civicrm.mo`.)

The only thing you might want to double-check (cc @mlutfy) is whether we need some corresponding in the POT/PO/Transifex world.

**Comment by mlutfy** [comment]
Fine by me. I added a note to this issue for the next string extraction: https://lab.civicrm.org/infra/ops/-/issues/1012

**Pull request merged**